clientv3/watch.go

```diff
@@ -46,6 +46,24 @@ type Watcher interface {
 	// through the returned channel. If revisions waiting to be sent over the
 	// watch are compacted, then the watch will be canceled by the server, the
 	// client will post a compacted error watch response, and the channel will close.
+	// If the context "ctx" is canceled or timed out, returned "WatchChan" is closed,
+	// and "WatchResponse" from this closed channel has zero events and nil "Err()".
+	// If the context is "context.Background/TODO", returned "WatchChan" will
+	// not be closed and wait until events happen, except when server returns
+	// a non-recoverable error and:
+	//  1. When context passed with "WithRequireLeader" and the connected server
+	//     has no leader, error "etcdserver: no leader" is returned, and then
+	//     "WatchChan" is closed with non-nil "Err()".
+	//  2. error is "grpc/*status.statusError" and its error code is neither
```

Comment:
isnt this covered by non-recoverable error? i do not want to list all errors here.

Reply:
@xiang90 Just removed that part. PTAL.